src/app/code/community/Omise/Gateway/Model/PaymentMethod.php

```diff
@@ -28,24 +25,21 @@ public function authorize(Varien_Object $payment, $amount)
     {
         Mage::log('Start authorize with OmiseCharge API!');
         
+        $order = $payment->getOrder();
         $omise_token = $payment->getData('additional_information');
-        $charge = Mage::getModel('omise_gateway/omiseCharge')->createOmiseCharge(array(
+        $charge = Mage::getModel('omise_gateway/omisecharge')->createOmiseCharge(array(
             "amount"        => number_format($amount, 2, '', ''),
-            "currency"      => "thb",
+            "currency"      => strtolower($order->getBaseCurrencyCode()),
```

Comment:
@duzydook We have a field for set currency in module setting page. So, is it possible to use that value instead of value from `getBaseCurrencyCode` function ? (what do you think ?)